abdul-moize/django_ecommerce
# Feat: Cart Management API

This PR adds Cart API  to main branch. Cart API is implemented in carts app. User can add products to cart, change a product in cart,remove a product from cart, view cart and clear the cart.

To Test this PR follow these steps:
1. Pull this branch and run: `pip install -r requirements.txt`
2. Run: `python manage.py migrate && python manage.py runserver`
3. Open postman, create an account and login and add product. see https://github.com/abdul-moize/django_ecommerce/pull/1 and https://github.com/abdul-moize/django_ecommerce/pull/2
- To test add to cart send a `post` request on `127.0.0.1:8000/carts/` 
![image](https://user-images.githubusercontent.com/88365652/133804186-91370463-951a-4887-90da-01fe36c02da0.png)
![image](https://user-images.githubusercontent.com/88365652/133804122-c4b75bc9-d9db-40fb-87ba-bd192fd7d796.png)

- To test update to cart send a `put` request on `127.0.0.1:8000/carts/<cart_item.id>/`
![image](https://user-images.githubusercontent.com/88365652/133804203-07ff0c89-4463-4a27-b863-c134d2f7e4ba.png)
![image](https://user-images.githubusercontent.com/88365652/133804377-c3ebf6f8-8ee9-4b21-9266-5e70603c1d31.png)

- To test remove from cart send a `delete` request on `127.0.0.1:8000/carts/<cart_item.id>/`
![image](https://user-images.githubusercontent.com/88365652/133804209-2438f794-d164-4700-967f-c8b134dee032.png) 
![image](https://user-images.githubusercontent.com/88365652/133804493-521d88fa-f9c2-4cc1-b2c5-d6e6db385255.png)

- To test view all carts send a `get` request on `127.0.0.1:8000/carts/` 
![image](https://user-images.githubusercontent.com/88365652/133804217-dfb835fe-c3ce-412f-8d66-18ed15990a83.png)
![image](https://user-images.githubusercontent.com/88365652/133804561-7e599af5-9fa2-4f5d-9266-95d996627739.png)

- To test view a cart send a `get`request on `127.0.0.1:8000/carts/<id>`
![image](https://user-images.githubusercontent.com/88365652/133804217-dfb835fe-c3ce-412f-8d66-18ed15990a83.png)
![image](https://user-images.githubusercontent.com/88365652/134021148-20d5d86d-13d4-476b-ae64-440679bcd02c.png)

- To test order submit send a `patch` request on `127.0.0.1:8000/carts/`
- To test clear cart send a `delete` request on `127.0.0.1:8000/carts/` 
![image](https://user-images.githubusercontent.com/88365652/133804223-ad1b005e-249b-423f-b2e5-d3fc404dee2b.png)
![image](https://user-images.githubusercontent.com/88365652/134024708-cc7a0e9c-f5a6-496d-abb6-959619e16703.png)
